nbviewer/providers/base.py

```diff
@@ -630,10 +630,21 @@ def filter_formats(self, nb, raw):
             except Exception as err:
                 app_log.info("failed to test %s: %s", self.request.uri, name)
 
+    # empty methods to be implemented by subclasses to make GET requests more modular
+    def format_notebook_request(self, **kwargs):
+        pass
```

Comment:
A docstring on both of these functions explaining what they should expect to receive in `kwargs` and what they are expected to do / return would be helpful.